static/app/views/dashboards/widgets/timeSeriesWidget/releaseBubbles/utils/createReleaseBuckets.spec.tsx

```diff
@@ -0,0 +1,160 @@
+import {createReleaseBuckets} from 'sentry/views/dashboards/widgets/timeSeriesWidget/releaseBubbles/utils/createReleaseBuckets';
+
+function createTimeSeries(numItems: number, intervalMs = 1000, start = new Date()) {
+  const timeSeries = [
+    {
+      data: Array.from(Array(numItems)).map((_, i) => ({
+        value: 0, // don't care about value
+        timestamp: new Date(start.getTime() + i * intervalMs).toISOString(),
+      })),
+    },
+  ];
+
+  // find min/max timestamp of *all* timeSeries
+  let minTime: Date | undefined;
+  let maxTime: Date | undefined;
+
+  for (const currentSeries of timeSeries) {
+    if (currentSeries.data.length < 2) {
+      continue;
+    }
+
+    const firstData = currentSeries.data[0];
+    const lastData = currentSeries.data[currentSeries.data.length - 1];
+    // I hope `data` is sorted
+    if (!minTime || new Date(firstData!.timestamp) < minTime) {
+      minTime = new Date(firstData!.timestamp);
+    }
+    if (!maxTime || new Date(lastData!.timestamp) > maxTime) {
+      maxTime = new Date(lastData!.timestamp);
+    }
+  }
+
+  return {minTime: minTime?.getTime(), maxTime: maxTime?.getTime()};
+}
```

Comment:
I'm confused, why go through the whole rigamarole of creating every bucket, if you only need the min and max time? Why not just get the max using multiplication?

Reply:
I'll clean this up, leftover from when we had `timeSeries`

Reply:
[`7fee5b5` (#85270)](https://github.com/getsentry/sentry/pull/85270/commits/7fee5b54e3a27c21076cbb73123cac1f33447bbd)